examples/http-filter-mod-req-header.yaml

```diff
@@ -0,0 +1,52 @@
+kind: GatewayClass
```

Comment:
What is the purpose of using `mod` in the filename?

Reply:
I used `mod-req-header` in the filename because I plan on having several filter examples. I'll rename the example to `http-filter.yaml` and can rename when I submit the subsequent PR that provides additional filter examples.